### PR TITLE
build: test enterprise features in the every-pr teamcity build

### PR DESCRIPTION
**Pull request by danhhz**
This is done by grabbing a one hour testing license from the
registration server.

Release note: None

### CODE REVIEW

**Review by tbg — APPROVED** [review]
:lgtm:

Do the PMs know that automated processes are hitting the endpoint (or is that the point in the first place)?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/34174#-:-LWuhJNH2HAyQeIo5etq:b18x3br)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dt)

File: build/teamcity-local-roachtest.sh [hunk 1]
```diff
@@ -18,10 +18,13 @@ run build/builder.sh make bin/roachprod bin/workload bin/roachtest
 tc_end_block "Compile roachprod/workload/roachtest"
 
 tc_start_block "Run local roachtests"
+# Grab a testing license good for one hour.
+COCKROACH_DEV_LICENSE=$(curl "https://register.cockroachdb.com/api/prodtest")
```
Comment (dt) [inline build/teamcity-local-roachtest.sh]: Do we want to look at the exit status of curl and/or sanity check the output at all?

**Review by dt — APPROVED** [review]

**Review by danhhz — COMMENTED** [review]
> Do the PMs know that automated processes are hitting the endpoint (or is that the point in the first place)?

This is a shiny new endpoint built just for this. I don't think it'll show up in anything we track, but @dt would know for sure.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/34174#-:-LWvKa3K9Pq8Sc_C-IQB:bhahskh)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @dt)

___
*[build/teamcity-local-roachtest.sh, line 22 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/34174#-LWs8DwIwptzvXkqRDG_-r1-22:-LWvKawW60uUjoek5wPw:b-oo7tip) ([raw file](https://github.com/cockroachdb/cockroach/blob/bc37387e6bf49b36718c3e2aae625d693bf8539d/build/teamcity-local-roachtest.sh#L22)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

Do we want to look at the exit status of curl and/or sanity check the output at all? 
</blockquote></details>

Some part of the `set -euo pipefail` bit above handles looking at the exit code, but turns out I needed a `-f` flag on curl to tell it to return a non-zero exit code on non-successful http codes. Done (also moved it earlier in the script, so we throw away less work if this ever does happen).

**Comment by danhhz** [comment]
bors r=dt,tbg

**Review by dt — COMMENTED** [review]
These requests will go into the Prometheus top line request count but not much else -- they don't write to the DB so unlikely it'd show up in anything product looks at.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/34174#-:-LWvSp_r8cHjpEnQX9LW:b-6sp91t)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

**Pull request merged**